### DISCUSSION TITLE
How to filter and remove only the unregistered torrents in transmission-remote?

### DISCUSSION THREAD

**Discussion by ghost** [transmission/transmission#4328] [Q&A]
I want *only* filter the (private) torrents which the tracker reports as unregistered. I would then automate the removal process, clearing the redundant torrents, potentially with their data (`--remove-and-delete`).

I know how to return the torrents which are in a stopped state (not a reliable method, some I've paused myself or removed their data without removing the torrents from the client):

    transmission-remote --auth user:password --list | grep "Stopped"

Is it possible to apply the `--info` flag to all torrents (`--list`)?

**Comment by opeeba** [comment]
I was looking for a better method than my script below and happened across this post. Hopefully it will be useful until I find a better way. Currently it searches for strings in the tracker status that various trackers use when they remove torrents. ("unregistered|error: complete|other: see|problem with file|dupe|does not exist").
I prefer to have it stop torrents, but you can edit it to use --remove-and-delete.

I have it setup to run hourly in cron.
```
transmission:~$ crontab -l
@hourly /var/lib/transmission/stop_unregistered_torrents.sh > /dev/null
```
```
#!/bin/ash

# Stop all unregistered torrent files

. .env

UNREGISTERED_TORRENTS=$(transmission-remote --auth "$TRANSMISSION_USER:$TRANSMISSION_PASSWORD" -tall -i | grep -E -B20 -i "unregistered|error: complete|other: see|problem with file|dupe|does not exist" | grep -i "id:" | tr -d [:blank:][Id:] | tr "\n" "," | sed 's/.$//')

if [ -z "$UNREGISTERED_TORRENTS" ]; then
  # No unregistered torrents found
  exit
else
  # Stop the unregistered torrents
  transmission-remote --auth "$TRANSMISSION_USER:$TRANSMISSION_PASSWORD" -t "$UNREGISTERED_TORRENTS" --stop
fi
```

**Reply by 13werwolf13** [reply]
You can also add the line `Torrent not registered`, I see it often.
